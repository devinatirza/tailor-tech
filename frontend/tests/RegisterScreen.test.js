import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import RegisterScreen from '../pages/Register';
import axios from 'axios';

jest.mock('axios');

describe('RegisterScreen', () => {
  it('test_handleRegister_successfulRegistration', async () => {
    const mockNavigation = { navigate: jest.fn() };
    const { getByText } = render(<RegisterScreen navigation={mockNavigation} />);
    const registerButton = getByText('Register');

    axios.post.mockResolvedValue({ data: { message: 'Registration successful' } });

    fireEvent.press(registerButton);

    await new Promise(process.nextTick); // Wait for promises to resolve

    expect(axios.post).toHaveBeenCalledWith("http://localhost:8000/register", expect.any(Object));
    expect(getByText('Registration successful')).toBeTruthy();
  });

  it('test_handleRegister_apiFailure', async () => {
    const mockNavigation = { navigate: jest.fn() };
    const { getByText } = render(<RegisterScreen navigation={mockNavigation} />);
    const registerButton = getByText('Register');

    axios.post.mockRejectedValue({ response: { data: { error: 'Failed to register' } } });

    fireEvent.press(registerButton);

    await new Promise(process.nextTick); // Wait for promises to resolve

    expect(axios.post).toHaveBeenCalledWith("http://localhost:8000/register", expect.any(Object));
    expect(getByText('Failed to register')).toBeTruthy();
  });

  it('test_handlePasswordChange_validations', () => {
    const { getByPlaceholderText } = render(<RegisterScreen navigation={{ navigate: jest.fn() }} />);
    const passwordInput = getByPlaceholderText('Password');

    fireEvent.changeText(passwordInput, 'Pass123!');

    expect(passwordInput.props.value).toBe('Pass123!');
    expect(getByPlaceholderText('Password Requirements:')).toBeTruthy();
    expect(getByText('Mininum 8 characters')).toBeTruthy();
    expect(getByText('At least one uppercase and lowercase letter')).toBeTruthy();
    expect(getByText('At least one number and special character')).toBeTruthy();
  });
});