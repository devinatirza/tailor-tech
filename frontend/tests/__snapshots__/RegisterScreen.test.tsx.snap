// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen renders register screen correctly (snapshot) 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F8F8F8",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 40,
    }
  }
>
  <Text
    style={
      {
        "color": "#401201",
        "fontSize": 28,
        "fontWeight": "bold",
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Welcome to TailorTech!
  </Text>
  <View
    style={
      {
        "marginBottom": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "borderBottomColor": "#401201",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "width": "100%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Name"
        style={
          {
            "fontSize": 16,
            "height": 50,
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#401201",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "width": "100%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Email"
        style={
          {
            "fontSize": 16,
            "height": 50,
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#401201",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "width": "100%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          {
            "fontSize": 16,
            "height": 50,
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 0,
          "marginTop": 0,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
        }
      }
    >
      Password Requirements:
    </Text>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "gap": 0,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <input
          checked={false}
          disabled={true}
          style={
            {
              "color": "#C1AEA7",
            }
          }
          type="checkbox"
        />
        <Text
          style={
            {
              "fontSize": 16,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
            }
          }
        >
          Mininum 8 characters
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <input
          checked={false}
          disabled={true}
          style={
            {
              "color": "#C1AEA7",
            }
          }
          type="checkbox"
        />
        <Text
          style={
            {
              "fontSize": 16,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
            }
          }
        >
          At least one uppercase and lowercase letter
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <input
          checked={false}
          disabled={true}
          style={
            {
              "color": "#C1AEA7",
            }
          }
          type="checkbox"
        />
        <Text
          style={
            {
              "fontSize": 16,
              "marginBottom": 0,
              "marginTop": 0,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
            }
          }
        >
          At least one number and special character
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#401201",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "width": "100%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Confirm Password"
        secureTextEntry={true}
        style={
          {
            "fontSize": 16,
            "height": 50,
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#401201",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "width": "100%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Phone Number"
        style={
          {
            "fontSize": 16,
            "height": 50,
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#401201",
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "width": "100%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Address"
        style={
          {
            "fontSize": 16,
            "height": 50,
            "paddingHorizontal": 10,
          }
        }
        value=""
      />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#D9C3A9",
        "borderRadius": 5,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "width": "60%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#260101",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Register
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Already have an account?
    </Text>
  </View>
</View>
`;
